neqo-http3/src/server.rs

```diff
@@ -113,18 +113,46 @@ impl Http3Server {
         self.server.ech_config()
     }
 
-    pub fn process(&mut self, dgram: Option<&Datagram>, now: Instant) -> Output {
+    pub fn process_into_buffer<'a>(
+        &mut self,
+        dgram: Option<Datagram<&[u8]>>,
+        now: Instant,
+        out: &'a mut Vec<u8>,
+    ) -> Output<&'a [u8]> {
         qtrace!([self], "Process.");
-        let out = self.server.process(dgram, now);
+        let mut output = self
+            .server
+            // TODO: NLL borrow issue. See https://github.com/rust-lang/rust/issues/54663
+            //
+            // Find alternative.
+            .process_into_buffer(dgram, now, unsafe { &mut *std::ptr::from_mut(out) });
```

Comment:
Anything left to do here?

Reply:
I have not been able to come up with a proper solution (i.e. no `unsafe`). Before merging here, I want to spend more time on it, unless of course, we go in a direction that no longer requires the trick in the first place.